From a7eef64f82347a74c9939776782bb55baeef6761 Mon Sep 17 00:00:00 2001
From: Pierre Riteau <priteau@uchicago.edu>
Date: Tue, 19 Sep 2017 14:47:58 +0100
Subject: [PATCH 2/3] Disable hash ring

---
 nova/virt/ironic/driver.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/nova/virt/ironic/driver.py b/nova/virt/ironic/driver.py
index 784674b7a5..d80388e1a4 100644
--- a/nova/virt/ironic/driver.py
+++ b/nova/virt/ironic/driver.py
@@ -543,13 +543,19 @@ class IronicDriver(virt_driver.ComputeDriver):
             return False
 
     def _refresh_hash_ring(self, ctxt):
+        # NOTE(priteau): We disable the hash ring for Chameleon since we run a
+        # nova-compute per hypervisor
+        '''
         service_list = objects.ServiceList.get_all_computes_by_hv_type(
             ctxt, self._get_hypervisor_type())
+        '''
         services = set()
+        '''
         for svc in service_list:
             is_up = self.servicegroup_api.service_is_up(svc)
             if is_up:
                 services.add(svc.host)
+        '''
         # NOTE(jroll): always make sure this service is in the list, because
         # only services that have something registered in the compute_nodes
         # table will be here so far, and we might be brand new.
-- 
2.16.1

